modules/prebidServerBidAdapter.js

```diff
@@ -339,6 +339,13 @@ const LEGACY_PROTOCOL = {
             if (bidObj.nurl) {
               bidObject.vastUrl = bidObj.nurl;
             }
+            // when video bid is already cached by Prebid Server, videoCacheKey and vastUrl should be provided properly
+            if (bidObj.cache_id) {
+              bidObject.videoCacheKey = bidObj.cache_id;
+            }
+            if (bidObj.cache_url) {
+              bidObject.vastUrl = bidObj.cache_url;
```

Comment:
I don't think this does anything by itself? ie, prebid.js will still try to cache this URL, so essentially your double caching here I believe. You'd need a way to signal to the core prebid library not to cache this bid.

Reply:
upon further review: it does check:
https://github.com/prebid/Prebid.js/blob/master/src/auction.js#L230

Reply:
I think now we are checking whether `videoCacheKey` is available when there's a `cache.url`: https://github.com/prebid/Prebid.js/blob/717b4391d641f14c3875620b06f3e1a8a4e827fb/src/auction.js#L230

Reply:
Do note that the current version of AMP support does not include `cache_url`, it only sends the `cache_id`.

Reply:
if that is the case, we need to update this to check both keys exist. Otherwise we'll break caching if `cache_url` doesn't exist.